src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -619,8 +622,10 @@ private[formal] object RepeatingInterval {
     case IsoFields.QUARTER_YEARS =>
       ldt.withMonth((ldt.getMonthValue - 1) / 4 + 1).withDayOfMonth(1).truncatedTo(ChronoUnit.DAYS)
     case ChronoUnit.MONTHS => ldt.withDayOfMonth(1).truncatedTo(ChronoUnit.DAYS)
-    case ChronoUnit.WEEKS =>
-      ldt.withDayOfYear(ldt.getDayOfYear - ldt.getDayOfWeek.getValue + 1).truncatedTo(ChronoUnit.DAYS)
+    case ChronoUnit.WEEKS => ldt.getDayOfYear - ldt.getDayOfWeek.getValue compare 0 match {
```

Comment:
Seems like this could be something simpler like
```
ldt.withDayOfYear((ldt.getDayOfYear - ldt.getDayOfWeek.getValue + 1) max 1).truncatedTo(ChronoUnit.DAYS)
```
Would that achieve the same thing?